### ISSUE TITLE
Replace php-pear drush installation with puppet 4.x logic

### ISSUE THREAD

**Issue by jeff1evesque** [jeff1evesque/drupal-demonstration#209]
The following from `default.pp`:

```puppet
...
$drush_console_table = 'Console_Table-1.1.5'
...
## install drush
exec {'install-drush':
    command => 'pear channel-discover pear.drush.org && pear install drush/drush',
    refreshonly => true,
    notify => Exec['install-drush-dependency'],
}

## drush requirement: need to install Console_Table-x.x.x
exec {'install-drush-dependency':
    command => "pear install ${drush_console_table}",
    refreshonly => true,
    notify => Exec['allow-htaccess-1'],
}
...
```

need to be replaced with another dedicated puppet manifest, with the following [installation](http://docs.drush.org/en/master/install/) process:

```bash
# Download latest stable release manually or use code below. See github.com/drush-ops/drush/releases.
wget https://github.com/drush-ops/drush/releases/download/8.0.0-rc4/drush.phar

# Test your install.
php drush.phar core-status

# Rename to `drush` instead of `php drush.phar`. Destination can be anywhere on $PATH. 
chmod +x drush.phar
sudo mv drush.phar /usr/local/bin/drush

# Enrich the bash startup file with completion and aliases.
drush init
```

The motivation for this issue is because [pear](http://pear.php.net/) is not intending to support drush, and it [may](https://www.drupal.org/node/1322564) have been discontinued from support, as mentioned by numerous folks on the IRC #drupal channel.

**Comment by jeff1evesque** [comment]
We may need to [restructure](http://www.standalone-sysadmin.com/blog/) the directory structure, corresponding to our puppet (4.x) implementation:

>The solution is to redesign your Vagrant puppet directory. Instead of “puppet/init.pp” and “puppet/modules/”, you need to make “puppet/environments/site/environmentname/manifests/site.pp” and “puppet/environment/site/environmentname/modules/”, respectively

**Comment by jeff1evesque** [comment]
We may need to [consider](https://github.com/mitchellh/vagrant/issues/3740#issuecomment-138770615) using puppet environments, instead of manifest file settings.

**Comment by jeff1evesque** [comment]
Currently, our traceback yields the following error:

```bash
...
==> default: Notice: /Stage[main]/Main/Exec[remove-rpm-package]: Triggered 'refresh' from 1 events
==> default: Notice: /Stage[main]/Main/Exec[update-php-1]: Triggered 'refresh' from 1 events
==> default: Notice: /Stage[main]/Main/Exec[mv-remi-repo-1]: Triggered 'refresh' from 1 events
==> default: Notice: /Stage[main]/Main/Exec[update-php-2]: Triggered 'refresh' from 1 events
==> default: Notice: /Stage[main]/Main/Exec[mv-epel-repo-2]: Triggered 'refresh' from 1 events
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: Loaded plugins: fastestmirror
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: Loading mirror speeds from cached hostfile
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns:  * base: centos.sonn.com
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns:  * epel: mirror.steadfast.net
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns:  * extras: repo.dimenoc.com
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns:  * remi: remirpm.mirror.gymkl.ch
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns:  * remi-php55: remirpm.mirror.gymkl.ch
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns:  * remi-php56: remirpm.mirror.gymkl.ch
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns:  * remi-safe: remirpm.mirror.gymkl.ch
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns:  * remi-test: remirpm.mirror.gymkl.ch
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns:  * updates: centos.chi.host-engine.com
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: Resolving Dependencies
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Running transaction check
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php.x86_64 0:5.4.16-36.el7_1 will be updated
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php.x86_64 0:5.6.16-1.el6.remi will be an update
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: httpd-mmn = 20051115 for package: php-5.6.16-1.el6.remi.x86_64
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-bcmath.x86_64 0:5.4.16-36.el7_1 will be updated
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-bcmath.x86_64 0:5.6.16-1.el6.remi will be an update
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-cli.x86_64 0:5.4.16-36.el7_1 will be updated
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-cli.x86_64 0:5.6.16-1.el6.remi will be an update
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-common.x86_64 0:5.4.16-36.el7_1 will be updated
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: php-json for package: phpMyAdmin-4.5.3.1-1.el6.remi.noarch
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: php-json for package: php-tcpdf-6.2.12-1.el6.remi.noarch
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: php-zip for package: phpMyAdmin-4.5.3.1-1.el6.remi.noarch
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: php-pecl-zip(x86-64) for package: php-common-5.6.16-1.el6.remi.x86_64
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-common.x86_64 0:5.6.16-1.el6.remi will be an update
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-gd.x86_64 0:5.4.16-36.el7_1 will be updated
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-gd.x86_64 0:5.6.16-1.el6.remi will be an update
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: gd-last(x86-64) >= 2.1.1 for package: php-gd-5.6.16-1.el6.remi.x86_64
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: libvpx.so.1()(64bit) for package: php-gd-5.6.16-1.el6.remi.x86_64
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: libpng12.so.0()(64bit) for package: php-gd-5.6.16-1.el6.remi.x86_64
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: libgd.so.3()(64bit) for package: php-gd-5.6.16-1.el6.remi.x86_64

==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-mbstring.x86_64 0:5.4.16-36.el7_1 will be updated
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-mbstring.x86_64 0:5.6.16-1.el6.remi will be an update
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-mysql.x86_64 0:5.4.16-36.el7_1 will be obsoleted
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-mysqlnd.x86_64 0:5.6.16-1.el6.remi will be obsoleting
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-pdo.x86_64 0:5.4.16-36.el7_1 will be updated
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-pdo.x86_64 0:5.6.16-1.el6.remi will be an update
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-process.x86_64 0:5.4.16-36.el7_1 will be updated
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-process.x86_64 0:5.6.16-1.el6.remi will be an update
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-tcpdf.noarch 0:6.2.11-1.el7 will be updated
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-tcpdf.noarch 0:6.2.12-1.el6.remi will be an update
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-tcpdf-dejavu-sans-fonts.noarch 0:6.2.11-1.el7 will be updated
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-tcpdf-dejavu-sans-fonts.noarch 0:6.2.12-1.el6.remi will be an update
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-tidy.x86_64 0:5.4.16-3.el7 will be updated
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-tidy.x86_64 0:5.6.16-1.el6.remi will be an update
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-xml.x86_64 0:5.4.16-36.el7_1 will be updated
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-xml.x86_64 0:5.6.16-1.el6.remi will be an update
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package phpMyAdmin.noarch 0:4.4.15.2-1.el7 will be updated
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package phpMyAdmin.noarch 0:4.5.3.1-1.el6.remi will be an update
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: php-composer(udan11/sql-parser) >= 3.0.8 for package: phpMyAdmin-4.5.3.1-1.el6.remi.noarch
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: php-composer(phpseclib/phpseclib) >= 2.0.0 for package: phpMyAdmin-4.5.3.1-1.el6.remi.noarch
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: php-recode for package: phpMyAdmin-4.5.3.1-1.el6.remi.noarch
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Running transaction check
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package gd-last.x86_64 0:2.1.1-2.el6.remi will be installed
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: libtiff.so.3()(64bit) for package: gd-last-2.1.1-2.el6.remi.x86_64
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package libpng12.x86_64 0:1.2.50-7.el7_2 will be installed
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package libvpx.x86_64 0:1.3.0-5.el7_0 will be installed
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php.x86_64 0:5.6.16-1.el6.remi will be an update
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: httpd-mmn = 20051115 for package: php-5.6.16-1.el6.remi.x86_64
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-pecl-jsonc.x86_64 0:1.3.9-1.el6.remi.5.6 will be installed
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-pecl-zip.x86_64 0:1.13.1-1.el6.remi.5.6 will be installed
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: libzip.so.4()(64bit) for package: php-pecl-zip-1.13.1-1.el6.remi.5.6.x86_64
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-phpseclib.noarch 0:2.0.0-4.el7 will be installed
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: php-gmp for package: php-phpseclib-2.0.0-4.el7.noarch
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: php-composer(symfony/class-loader) for package: php-phpseclib-2.0.0-4.el7.noarch
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-recode.x86_64 0:5.6.16-1.el6.remi will be installed
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: librecode.so.0()(64bit) for package: php-recode-5.6.16-1.el6.remi.x86_64
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-udan11-sql-parser.noarch 0:3.0.8-1.el6.remi will be installed
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Running transaction check
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package compat-libtiff3.x86_64 0:3.9.4-11.el7 will be installed
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package libzip-last.x86_64 0:1.0.1-1.el6.remi will be installed
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php.x86_64 0:5.6.16-1.el6.remi will be an update
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: httpd-mmn = 20051115 for package: php-5.6.16-1.el6.remi.x86_64
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-gmp.x86_64 0:5.6.16-1.el6.remi will be installed
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: libgmp.so.3()(64bit) for package: php-gmp-5.6.16-1.el6.remi.x86_64
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-symfony-class-loader.noarch 0:2.7.8-1.el6.remi will be installed
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: php-symfony-common = 2.7.8-1.el6.remi for package: php-symfony-class-loader-2.7.8-1.el6.remi.noarch
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package recode.x86_64 0:3.6-38.el7 will be installed
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Running transaction check
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php.x86_64 0:5.6.16-1.el6.remi will be an update
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: httpd-mmn = 20051115 for package: php-5.6.16-1.el6.remi.x86_64
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-gmp.x86_64 0:5.6.16-1.el6.remi will be installed
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Processing Dependency: libgmp.so.3()(64bit) for package: php-gmp-5.6.16-1.el6.remi.x86_64
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: ---> Package php-symfony-common.noarch 0:2.7.8-1.el6.remi will be installed
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: --> Finished Dependency Resolution
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: Error: Package: php-gmp-5.6.16-1.el6.remi.x86_64 (remi-php56)
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: Requires: libgmp.so.3()(64bit)
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: Error: Package: php-5.6.16-1.el6.remi.x86_64 (remi-php56)
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: Requires: httpd-mmn = 20051115
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: Installed: httpd-2.4.6-40.el7.centos.x86_64 (@base)
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: httpd-mmn = 20120211
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: httpd-mmn = 20120211x8664
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns: httpd-mmn = 20120211-x86-64
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns:  You could try using --skip-broken to work around the problem
==> default: Notice: /Stage[main]/Main/Exec[update-yum-php]/returns:  You could try running: rpm -Va --nofiles --nodigest
==> default: Error: /Stage[main]/Main/Exec[update-yum-php]: Failed to call refresh: yum -y update returned 1 instead of one of [0]
==> default: Error: /Stage[main]/Main/Exec[update-yum-php]: yum -y update returned 1 instead of one of [0]
==> default: Notice: /Stage[main]/Main/Exec[restart-services]: Dependency Package[php-opcache] has failures: true
==> default: Error: Execution of '/bin/yum -d 0 -e 0 -y install php-opcache' returned 1: Error: Package: php-gmp-5.6.16-1.el6.remi.x86_64 (remi-php56)
==> default:            Requires: libgmp.so.3()(64bit)
==> default: Error: Package: php-5.6.16-1.el6.remi.x86_64 (remi-php56)
==> default:            Requires: httpd-mmn = 20051115
==> default:            Installed: httpd-2.4.6-40.el7.centos.x86_64 (@base)
==> default:                httpd-mmn = 20120211
==> default:                httpd-mmn = 20120211x8664
==> default:                httpd-mmn = 20120211-x86-64
==> default: Error: Package: php-mysql-5.4.45-2.el6.remi.x86_64 (remi)
==> default:            Requires: php-pdo(x86-64) = 5.4.45-2.el6.remi
==> default:            Removing: php-pdo-5.4.16-36.el7_1.x86_64 (@base)
==> default:                php-pdo(x86-64) = 5.4.16-36.el7_1
==> default:            Updated By: php-pdo-5.6.16-1.el6.remi.x86_64 (remi-php56)
==> default:                php-pdo(x86-64) = 5.6.16-1.el6.remi
==> default:            Available: php-pdo-5.4.45-1.el6.remi.x86_64 (remi)
==> default:                php-pdo(x86-64) = 5.4.45-1.el6.remi
==> default:            Available: php-pdo-5.4.45-2.el6.remi.x86_64 (remi)
==> default:                php-pdo(x86-64) = 5.4.45-2.el6.remi
==> default:            Available: php-pdo-5.5.29-1.el6.remi.x86_64 (remi-php55)

==> default:                php-pdo(x86-64) = 5.5.29-1.el6.remi
==> default:            Available: php-pdo-5.5.30-1.el6.remi.x86_64 (remi-php55)

==> default:                php-pdo(x86-64) = 5.5.30-1.el6.remi
==> default:            Available: php-pdo-5.6.15-1.el6.remi.x86_64 (remi-php56)

==> default:                php-pdo(x86-64) = 5.6.15-1.el6.remi
==> default:  You could try using --skip-broken to work around the problem
==> default:  You could try running: rpm -Va --nofiles --nodigest
==> default: Error: /Stage[main]/Main/Package[php-opcache]/ensure: change from purged to present failed: Execution of '/bin/yum -d 0 -e 0 -y install php-opcache' returned 1: Error: Package: php-gmp-5.6.16-1.el6.remi.x86_64 (remi-php56)
==> default:            Requires: libgmp.so.3()(64bit)
==> default: Error: Package: php-5.6.16-1.el6.remi.x86_64 (remi-php56)
==> default:            Requires: httpd-mmn = 20051115
==> default:            Installed: httpd-2.4.6-40.el7.centos.x86_64 (@base)
==> default:                httpd-mmn = 20120211
==> default:                httpd-mmn = 20120211x8664
==> default:                httpd-mmn = 20120211-x86-64
==> default: Error: Package: php-mysql-5.4.45-2.el6.remi.x86_64 (remi)
==> default:            Requires: php-pdo(x86-64) = 5.4.45-2.el6.remi
==> default:            Removing: php-pdo-5.4.16-36.el7_1.x86_64 (@base)
==> default:                php-pdo(x86-64) = 5.4.16-36.el7_1
==> default:            Updated By: php-pdo-5.6.16-1.el6.remi.x86_64 (remi-php56)
==> default:                php-pdo(x86-64) = 5.6.16-1.el6.remi
==> default:            Available: php-pdo-5.4.45-1.el6.remi.x86_64 (remi)
==> default:                php-pdo(x86-64) = 5.4.45-1.el6.remi
==> default:            Available: php-pdo-5.4.45-2.el6.remi.x86_64 (remi)
==> default:                php-pdo(x86-64) = 5.4.45-2.el6.remi
==> default:            Available: php-pdo-5.5.29-1.el6.remi.x86_64 (remi-php55)

==> default:                php-pdo(x86-64) = 5.5.29-1.el6.remi
==> default:            Available: php-pdo-5.5.30-1.el6.remi.x86_64 (remi-php55)

==> default:                php-pdo(x86-64) = 5.5.30-1.el6.remi
==> default:            Available: php-pdo-5.6.15-1.el6.remi.x86_64 (remi-php56)

==> default:                php-pdo(x86-64) = 5.6.15-1.el6.remi
==> default:  You could try using --skip-broken to work around the problem
==> default:  You could try running: rpm -Va --nofiles --nodigest
==> default: Warning: /Stage[main]/Main/Exec[restart-services]: Skipping because of failed dependencies
==> default: Notice: Applied catalog in 259.71 seconds
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

**Comment by jeff1evesque** [comment]
Since we are unable to implement the `wget::fetch` implementation, we will manually `wget` the necessary drush components.
